Refactor

bitionaire opened the pull request:
* Moved and renamed classes
* Replaced Exception with `RuntimeException` (easier to use, if the users know what they're doing)
* Removed Logger (easier for users to implement this library without too many dependencies)
* Tweaks here and there

What do you think?

Germandrummer92 commented:
Everything seems good! 

What does the change in the XMLParser Interface from <T extends XMLContext> to XMLContext as the return type actually change? Just for my own understanding what's the actual difference between the two?

bitionaire commented:
Just renamed the classes, as the former `XPathParser` might be a more general `XmlParser` and the name `XmlContext` is maybe clearer than `Context` in general.

The return types changed slightly as I removed the `Context` generic parameter from the parser. So now it's not possible to return a specific `Context` implementation. I used this to fix the value returning issues, as I return two different `Context` implemenations (Vtd & Value).

Besides I added the method `asFormattedString()` to the `Context` interface. Yet I'm not sure whether it's better to use a boolean parameter in `asString()` to indicate whether the response should be formatted. We could add a default implementation to the interface, e.g.

```java
interface XmlContext {

   // .. other methods

   String asString();

   default String asString(boolean formatted) {
      if (formatted) return asString()
      return prettyFormat(asString());
   }

   private static String prettyFormat(String string) {
      // ... use implementation from VtdXmlParserContext here
   }
}
```
